### ISSUE TITLE
Better handeling of non full blocks

### ISSUE THREAD

**Issue by MR-Spagetty** [luccaPossamai/optical#32]
As it stands if a beam is being into a block like moss carpet from the side for example it does not go across even though it clearly should be able to

**Comment by MR-Spagetty** [comment]
i think you misunderstood what i am talking about, i am not talking about a binary can or cannot pass through the entire block i am talking about passing through spaces on blocks where it would be physically possible and not through other spots for example a hollow log block from any mod that adds it will typically be open at 2 opposing ends and the rest of the faces will be closed i am suggesting that the beam would be able to pass through these open faces and not the others or another example would be something like a slab or vertical slab the beam would be able to pass until it physically hits the slab rather than either fully passing through no matter what or not even entering the block space

**Comment by luccaPossamai** [comment]
I did understood what you're talking. There's so many ways to implement this penetrable feature, the way I did it was by verifying the block's transparency. First problem about this is just limited types of blocks that can be penetrable because there's blocks that depending on the face the light hits it can or cannot pass through as you said. But I really want to be customizable outside the hardcode environment

**Comment by FoxyFox909** [comment]
A data-driven approach with tags is the sanest way to do it honestly, which the mod is already doing :)

**Comment by luccaPossamai** [comment]
Yup, the idea is to keep this mod in the Minecraft style. I don't intend to mess with more sophisticated methods to simulate light computationally (I usually avoid that kind of headache).
